### ISSUE TITLE
bench.ts fences its timed pass by accident, and the timing is the deliverable there

### ISSUE THREAD

**Issue by felladrin** [felladrin/gguf-trainer#99]
`src/commands/bench.ts` times one pass per case like this:

```ts
/** One timed pass: forward (+ backward), then a fence. */
async function once(gpu: WebGPUBackend, c: Case) {
  for (const t of c.inputs) t.zeroGrad();
  const out = c.run(gpu);
  if (c.backward) { /* seed and record backward */ }
  await gpu.sync([]);
}
```

The comment says "then a fence", and today there is one. Not for the reason the call looks like, though. `sync([])` passes no reads, so the only thing that can stage is `touchedExternals`, and every case's inputs come from `randTensor`, which builds them with `Tensor.zeros(shape, true)`. They are `requiresGrad` externals, so their gradients stage, the `mapAsync` awaits are real, and the timing includes the GPU tail.

Change any of that and the fence disappears silently:

- a case whose inputs are frozen (`requiresGrad` false), which is what a forward-only or LoRA-shaped case would naturally use;
- a case that calls `keepGradOnDevice`, which is what the resident training loop does and what a bench case modelling it would copy;
- an input that never becomes a touched external.

Any of those empties `stagings`, and #87 established what an empty `sync()` does: it submits and returns, awaiting nothing. The pass would then be timed to the submit rather than to completion, and the reported number would quietly stop including the GPU work. Every other place this matters produces a wrong result you can see; here it produces a plausible number, in the one command whose entire output is numbers.

This is not a live bug and nothing in the current bench is wrong. It is a load-bearing property with nothing holding it, in a file where the failure is silent.

Surfaced while reviewing #98.

Two ways to close it, and picking is the work:

- **Fence explicitly.** `await gpu.queue.onSubmittedWorkDone()` after the sync, or an equivalent that does not depend on what the case's inputs happen to be. Costs a real drain per timed pass, which is the thing being timed, so it wants measuring against the current numbers before it is taken: if it moves them, the published table in `docs/optimization.md` moves with it.
- **Assert the property.** Keep the incidental fence and make it non-incidental: check `gpu.lastSyncReadbackBytes > 0` after the sync and fail the bench if a case ever stops staging. Cheap, and it turns a silent degradation into a loud one, but it pins the current shape rather than removing the dependency.

Whichever it is, the comment should say what actually makes the pass fenced, the way `sync()`'s own comment now does.
